Sources/Onboarding/Callouts/OnboardingPlaylistView.swift

```diff
@@ -0,0 +1,476 @@
+// Copyright 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import SwiftUI
+import BraveUI
+import DesignSystem
+import Lottie
+import Strings
+
+public class OnboardingPlaylistModel: ObservableObject {
+  public enum Step: Equatable {
+    case initial
+    case details
+    case tryItOut
+    case completed(folderName: String)
+  }
+  
+  public var step: Step {
+    get { _step }
+    set {
+      withAnimation(.spring(dampingFraction: 0.9)) {
+        _step = newValue
+      }
+    }
+  }
+  
+  @Published private var _step: Step
+  
+  public var onboardingCompleted: (() -> Void)?
+  
+  public init(initialStep: Step = .initial) {
+    self._step = initialStep
+  }
+}
+
+public struct OnboardingPlaylistView: View {
+  @ObservedObject var model: OnboardingPlaylistModel
+  
+  public init(model: OnboardingPlaylistModel) {
+    self.model = model
+  }
+  
+  @State private var introAnimationCompleted: Bool = false
+  
+  @Environment(\.dismiss) private var dismiss
+  
+  private func icon(named: String, x: CGFloat, y: CGFloat, size: CGFloat = 18) -> some View {
+    Image(braveSystemName: named)
```

Comment:
`decorative: ` arg should be added for this constructor or `accessibilityHidden` otherwise every single leo icon seems to be picked by the VoiceOver

Reply:
Fixed in [8ee2c95](https://github.com/brave/brave-ios/pull/7527/commits/8ee2c959201fb2f45c901d7ced7bca609a29160d)